From 3f638b527fc46625d9dc791a07eea573c66f5d3c Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 10 Apr 2017 21:02:53 +0200
Subject: Correct two libecal/client tests

The main loop should be passed always as a user_data, because
the function called from the callback expects it and there is
no guarantee in which order the callbacks will be called.

diff --git a/tests/libecal/client/test-cal-client-get-view.c b/tests/libecal/client/test-cal-client-get-view.c
index 47514cc..2280646 100644
--- a/tests/libecal/client/test-cal-client-get-view.c
+++ b/tests/libecal/client/test-cal-client-get-view.c
@@ -206,7 +206,7 @@ test_get_view_sync (ETestServerFixture *fixture,
 	g_signal_connect (view, "objects_added", G_CALLBACK (objects_added_cb), fixture->loop);
 	g_signal_connect (view, "objects_modified", G_CALLBACK (objects_modified_cb), fixture->loop);
 	g_signal_connect (view, "objects_removed", G_CALLBACK (objects_removed_cb), fixture->loop);
-	g_signal_connect (view, "complete", G_CALLBACK (complete_cb), NULL);
+	g_signal_connect (view, "complete", G_CALLBACK (complete_cb), fixture->loop);
 
 	e_cal_client_view_set_fields_of_interest (view, NULL, &error);
 	if (error)
diff --git a/tests/libecal/client/test-cal-client-revision-view.c b/tests/libecal/client/test-cal-client-revision-view.c
index 0e25114..fdb386c 100644
--- a/tests/libecal/client/test-cal-client-revision-view.c
+++ b/tests/libecal/client/test-cal-client-revision-view.c
@@ -250,7 +250,7 @@ test_get_revision_view_sync (ETestServerFixture *fixture,
 	g_signal_connect (view, "objects_added", G_CALLBACK (objects_added_cb), fixture->loop);
 	g_signal_connect (view, "objects_modified", G_CALLBACK (objects_modified_cb), fixture->loop);
 	g_signal_connect (view, "objects_removed", G_CALLBACK (objects_removed_cb), fixture->loop);
-	g_signal_connect (view, "complete", G_CALLBACK (complete_cb), NULL);
+	g_signal_connect (view, "complete", G_CALLBACK (complete_cb), fixture->loop);
 
 	field_list = g_slist_prepend (NULL, (gpointer) "UID");
 	field_list = g_slist_prepend (field_list, (gpointer) "RECURRENCE-ID");
